foundry.md

```diff
@@ -145,6 +157,73 @@ This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32)`
          </account>
 ```
 
+#### `expectRevert` - The next call is expected to fail
+
+In addition, cheatcodes should be ignored during the execution of the `expectRevert` cheatcode.
+
+```k
+    rule [call.expectRevert]:
+         <k> CALL _ CHEAT_ADDR 0 ARGSTART _ARGWIDTH 0 0 => 1 ~> #push ... </k>
+         <output> _ => .ByteArray </output>
+         <localMem> LM </localMem>
+         <expectRevert> false => true </expectRevert>
+         <expectedBytes> _ => .ByteArray </expectedBytes>
+      requires CHEAT_ADDR ==Int #address(FoundryCheat)
+       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 4102309908 // selector ( "expectRevert()" )
+      [priority(40)]
+
+    rule [call.expectRevertBytes]:
+         <k> CALL _ CHEAT_ADDR 0 ARGSTART _ARGWIDTH 0 0 => 1 ~> #push ... </k>
+         <output> _ => .ByteArray </output>
+         <localMem> LM </localMem>
+         <expectRevert> false => true </expectRevert>
+         <expectedBytes> _ => #range(LM, ARGSTART +Int 4, 32) </expectedBytes>
+      requires CHEAT_ADDR ==Int #address(FoundryCheat)
+       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 4069379763 // selector ( "expectRevert(bytes)" )
+      [priority(40)]
+
+    rule [call.expectRevertBytes4]:
+         <k> CALL _ CHEAT_ADDR 0 ARGSTART _ARGWIDTH 0 0 => 1 ~> #push ... </k>
+         <output> _ => .ByteArray </output>
+         <localMem> LM </localMem>
+         <expectRevert> false => true </expectRevert>
+         <expectedBytes> _ => #range(LM, ARGSTART +Int 4, 4) </expectedBytes>
+      requires CHEAT_ADDR ==Int #address(FoundryCheat)
+       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 3273568480 // selector ( "expectRevert(bytes4)" )
+      [priority(40)]
+
+    rule [call.setEndExpectRevert]:
+         <k> CALL _ ADDR _ _ _ _ _ ~> (. => #endExpectRevert) ~> KITEM ... </k>
+         <output> _ => .ByteArray </output>
+         <expectRevert> true </expectRevert>
+      requires ADDR =/=Int #address(FoundryCheat)
+       andBool KITEM =/=K #endExpectRevert
+      [priority(40)]
+
+    syntax KItem ::= "#endExpectRevert" [klabel(foundry_endExpectRevert)]
+ // ---------------------------------------------------------------------
+    rule <k> #endExpectRevert => . ... </k>
+         <expectRevert> true => false </expectRevert>
+         <expectedBytes> EXPECTED => .ByteArray </expectedBytes>
+         <output> OUT </output>
+         <statusCode> _:ExceptionalStatusCode </statusCode>
+      requires OUT ==K EXPECTED
+        orBool EXPECTED ==K .ByteArray
+
+    rule <k> #endExpectRevert => #assert(false) ... </k>
+         <expectRevert> true => false </expectRevert>
+         <expectedBytes> _ => .ByteArray </expectedBytes> [owise]
+```
+
+```k
+    rule [call.ignoreCheatcodes]:
+         <k> CALL _ CHEAT_ADDR 0 _ARGSTART _ARGWIDTH 0 0 => 1 ~> #push ... </k>
+         <output> _ => .ByteArray </output>
+         <expectRevert> true </expectRevert>
+      requires CHEAT_ADDR ==Int #address(FoundryCheat)
+      [priority(45)]
```

Comment:
This rule is to make sure that "After calling expectRevert, calls to other cheatcodes before the reverting call are ignored." right? Shouldn't the priority be lower than 40?